packages/eventual-send/src/index.d.ts

```diff
@@ -18,6 +18,20 @@ export type EOnly<T> = T extends (...args: infer P) => infer R
     >
   : ERef<T>;
 
+type FilteredKeys<T, U> = {
+  [P in keyof T]: T[P] extends U ? P : never;
+}[keyof T];
+
+type DataOnly<T> = Omit<T, FilteredKeys<T, Function>>;
+type FunctionOnly<T> = Pick<T, FilteredKeys<T, Function>> &
+  (T extends Function ? (...args: Parameters<T>) => ReturnType<T> : {});
+
+interface Remotable<T> {
+  __Remote__: T;
+}
+type Remote<Primary, Local = DataOnly<Primary>> = ERef<
+  Local & Remotable<Primary>
+>;
 type Unpromise<T> = T extends ERef<infer U> ? U : T;
 
 type Parameters<T> = T extends (...args: infer T) => any ? T : any;
```

Comment:
These are extraneous from `lib.es5.d.ts`

Reply:
Also while we're at it, I believe `Property === PropertyKey`

Reply:
For each of `Promise<T>`, `Awaited<T>`, or `Unpromise<T>`, what's their treatment of non-promise thenables?

Reply:
`PromiseLike<T>` is a thenable which resolves to `T`.

`Promise<PromiseLike<T>>` doesn't handle recursion and is distinct from 'T'.

`Unpromise<PromiseLike<T>>` strips off just one layer and is just `T`.  `Unpromise<PromiseLike<PromiseLIke<T>>` is `PromiseLike<T>`.

`Awaited<PromiseLike<T>>` recurses, and is just `T`.
`Awaited<PromiseLike<PromiseLike<T>>>` is just `T` as well.